src/publish.ts

```diff
@@ -0,0 +1,135 @@
+import { isMapDocumentNode, isProfileDocumentNode } from '@superfaceai/ast';
+import { isProviderJson } from '@superfaceai/one-sdk';
+import { parseMap, parseProfile, Source } from '@superfaceai/parser';
+import { ServiceApiError, ServiceClient } from '@superfaceai/service-client';
+
+import {
+  exists,
+  EXTENSIONS,
+  localMaps,
+  localProfiles,
+  localProviders,
+  PrintFn,
+  readFile,
+} from './util';
+
+export type PublishOptions = {
+  print?: PrintFn;
+  baseUrl?: string;
+  refreshToken?: string;
+};
+
+export async function publish(
+  path: string,
+  options?: PublishOptions
+): Promise<void> {
+  if (!(await exists(path))) {
+    throw new Error('Path does not exist');
+  }
+
+  const client = new ServiceClient({
+    baseUrl: options?.baseUrl,
+    refreshToken: options?.refreshToken,
+  });
+
+  const file = await readFile(path);
+
+  if (path.endsWith(EXTENSIONS.provider)) {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    const parsedFile = JSON.parse(file);
+    if (isProviderJson(parsedFile)) {
+      options?.print?.(`Publishing provider ${parsedFile.name} from: ${path}`);
+      await client.createProvider(file);
+    } else {
+      throw new Error('File does not have provider json structure');
+    }
+  } else if (path.endsWith(EXTENSIONS.profile)) {
+    const parsedFile = parseProfile(new Source(file, path));
+    if (isProfileDocumentNode(parsedFile)) {
+      options?.print?.(
+        `Publishing profile "${parsedFile.header.name}" from: ${path}`
+      );
+      await client.createProfile(file);
+    } else {
+      throw new Error('Unknown profile file structure');
+    }
+  } else if (path.endsWith(EXTENSIONS.map)) {
+    const parsedFile = parseMap(new Source(file, path));
+    if (isMapDocumentNode(parsedFile)) {
```

Comment:
Similarly as in provider, we could use `assertMapDocumentNode` or `assertProfileDocumentNode` from ast - better errors.

Reply:
See https://github.com/superfaceai/station/pull/73#discussion_r714521913